content/en/blog/2022-08-26-manage-kyverno-policies-as-ocirepositories/index.md

```diff
@@ -0,0 +1,195 @@
+url: /blog/2022/08/manage-kyverno-policies-as-ocirepositories
+resources:
+- src: "**.{png}"
+  title: "Image #:counter"
+---
+
+{{< imgproc twitter-ss Resize 400x >}}
+{{< /imgproc >}}
+
+The Flux team has released a new version of Flux v0.32 that includes fantastic features. One of them is OCI Repositories feature that allows us to store and distribute a wide variety of sources such as Kubernetes manifests, Kustomize overlays, and Terraform modules as OCI (Open Container Initiative) Artifacts. Furthermore, the Flux team got us even more excited because they are planning to verify the authenticity of the OCI Artifacts before they get applied into Kubernetes by integrating cosign, which is one of the most significant projects from the @projectsigstore community that help us to sign and verify OCI images, blobs, etc., please see the issue to get more detail about the plan.
+
+Today’s blog post is all about a quick tour of this feature and will give you a real-world example of it to show you how you can leverage this feature to manage Kyverno policies as OCI Artifacts. It is worth saying that this topic has been discussed for a while in the Kyverno community, too. There is an ongoing [issue](https://github.com/kyverno/KDP/pull/19) about packaging and distributing Kyverno policies as OCI Artifacts through its CLI. Also, there is a chance to move that logic into Kyverno’s core.
+
+But for those who might not be familiar enough with [OCI Artifacts](https://github.com/opencontainers/artifacts/) (including me), it’s worth explaining what the OCI Artifacts are before jumping into the details. OCI Artifacts gives you the power of storing and distributing other types of data (nearly anything), such as Kubernetes deployment files, [Helm Charts](https://helm.sh/), [and CNAB](https://cnab.io/), in addition to container images via OCI registries. And today, we’ll be using this feature for Kyverno policies. To be more precise, OCI Artifacts are not a new specification, format, or API. It just utilizes the existent [OCI manifest](https://github.com/opencontainers/image-spec/blob/master/manifest.md) and [OCI index](https://github.com/opencontainers/image-spec/blob/master/image-index.md) definitions. Hence, we can quickly start using the same client tooling, such as a crane, skopeo, etc., and distribute them using OCI registries, thanks to the [OCI distribution-spec](https://github.com/opencontainers/distribution-spec/). Because OCI Artifacts does not change anything related to the specs, it only expands them to give people (artifact authors) power to define their content types. It is more like a generic definition for determining what can be stored in an OCI registry and consumed by clients. At the time of writing this, the Flux team does not define their types, but I’ve opened an [issue](https://github.com/fluxcd/pkg/issues/319) requesting them to create ones, let’s see what will happen in this space, stay tuned.
```

Comment:
> At the time of writing this, the Flux team does not define their types, but I’ve opened an issue

You're presenting this like it's a bad thing. In RFC-0003 we've explained why having a Flux media type doesn't make much sense, an artifact can contain various configurations that are not specific to Flux (Kubernetes manifests, Terraform code, etc). The Flux project can't register a media type for Terraform or Kubernetes  e.g. `application/vnd.cncf.kubernetes.content.v1.tar+gzip`, it's up to these projects to own and register such a type with IANA.

Reply:
Thx for the explanation 🙏